projects/systelab-components/src/lib/image-viewer/image-viewer.component.ts

```diff
@@ -0,0 +1,377 @@
+
+			this.initializeCommonParameters(event.clientX, event.clientY);
+
+			if (this.dragEnabled) {
+				this.initializeDragAction();
+
+			} else if (this.zoomEnabled) {
+				this.initializeZoomByAreaAction();
+			}
+		}
+	}
+
+	@HostListener('mousemove', ['$event'])
+	public doMouseMove(event: MouseEvent): void {
+		if (this.imageClicked) {
+			if (this.dragEnabled) {
+
+				this.scrollViewport(this.zoomArea.cursorStartX - event.clientX + this.zoomArea.left,
+					this.zoomArea.cursorStartY - event.clientY + this.zoomArea.top);
+
+			} else if (this.zoomEnabled) {
+				this.updateZoomSelector(event.clientX, event.clientY);
+			}
+		}
+	}
+
+	@HostListener('mouseup')
+	public doMouseUp(): void {
+		if (this.imageClicked) {
+			if (this.dragEnabled) {
+				this.zoomArea.scrollTop = this.viewPort.scrollTop;
+				this.zoomArea.scrollLeft = this.viewPort.scrollLeft;
+
+			} else if (this.zoomEnabled && this.zoomSelector.allow) {
+				if (this.zoomSelector.width > 0 && this.zoomSelector.height > 0) {
+					this.resizeZoomSelectorAndImage();
+				}
+				this.chref.detectChanges();
+				this.imgParams.sliderZoomPct = this.getSliderPct();
+			}
+			this.deactivateZoomSelector();
+			this.imageClicked = false;
+		} else {
+			this.zoomArea.scrollTop = this.viewPort.scrollTop;
+			this.zoomArea.scrollLeft = this.viewPort.scrollLeft;
+		}
+	}
+
+	public getWidth(): string {
+		if (this.zoomEnabled) {
+			return this.imgParams.width + 'px';
+		} else {
+			return this.imgParams.sliderZoomPct ? this.image.naturalWidth * this.imgParams.sliderZoomPct / 100 + 'px'
+				: undefined;
+		}
+	}
+
+	public getFilter(): string {
+		return this.imgParams.filter;
+	}
+
+	public setFilter(filter: string): void {
+		this.imgParams.filter = filter;
+	}
+
+	public doAdjust(): void {
+		this.imgParams.sliderZoomPct = this.getInitialZoom();
+		this.dragEnabled = false;
+		this.zoomEnabled = false;
+	}
+
+	public toggleZoomByArea(): void {
+		if (this.zoomEnabled) {
+			this.zoomEnabled = false;
+			this.dragEnabled = this.imageOverflowViewport();
+		} else {
+			this.scaleImage(1);
+			this.zoomEnabled = true;
+			this.dragEnabled = false;
+		}
+	}
+
+	public sliderZoomChanged(): void {
+		this.dragEnabled = this.imageOverflowViewport();
+		this.zoomEnabled = false;
+		this.scaleImage(1);
+	}
+
+	public setInitialValues(): void {
+		this.viewPort = this.elementRef.nativeElement;
+		this.image = this.imageViewerImg.nativeElement;
+		this.wrapper = this.imageViewerWrapper.nativeElement;
+
+		// Set zoom scale width and marks
+		this.zoomScale.zoomScaleLength = this.viewPort.querySelector('input[type="range"]').offsetWidth;
+		this.zoomScale.zoomScaleChunk = Math.round(this.zoomScale.zoomScaleLength / 6);
+		// Set zoom to fit the image
+		this.imgParams.sliderZoomPct = this.getInitialZoom();
+		// Min zoom 10% smaller than initial zoom
+		this.sliderZoomMin = this.imgParams.sliderZoomPct - 10;
+		// Max zoom to double the real size of the image
+		this.sliderZoomMax = 200;
```

Comment:
It would be better to have an optional parameter to set the max zoom (default 200)

Reply:
It makes sense. I will review it.

Reply:
Updated. Now the maximum zoom is managed as an input parameter.